pkg/etcd/etcd.go

```diff
@@ -1116,6 +1163,48 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
+func (e *ETCD) setEtcdStatusCondition(node *v1.Node, promoted bool) (*v1.Node, error) {
+	etcdStatusType := v1.NodeConditionType("EtcdIsVoter")
+
+	var newCondition v1.NodeCondition
+	if promoted {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "True",
+			Reason:  "MemberIsLearner",
+			Message: "Node is a voting member of the etcd cluster",
+		}
+	} else {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "MemberNotLearner",
+			Message: "Node has not been promoted to voting member of the etcd cluster",
+		}
+	}
+
+	for i, condition := range node.Status.Conditions {
+		if i == (len(node.Status.Conditions)-1) && condition.Type != etcdStatusType {
+			logrus.Infof("Adding etcd member status condition")
+			node.Status.Conditions = append(node.Status.Conditions, newCondition)
+			break
+		}
+
+		if condition.Type == etcdStatusType && condition.Status == "True" {
```

Comment:
1. If the status changes, set `LastTransitionTime` to now.
2. Always update `LastHeartbeatTime`

See: https://github.com/kubernetes/kubernetes/blob/v1.28.3/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L871-L898